### PR TITLE
Finalise Java 7 downgrade

### PR DESCRIPTION
**Pull request by AndyMacDroo**
Closes https://github.com/dmi3coder/behaiv-java/issues/15

- Create DataMappingUtils to hold new collections mapping methods. 
- Add relevant tests for the new utility class.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -14,8 +14,8 @@ plugins {
     id 'maven-publish'
 }
 
-sourceCompatibility = 1.8
-targetCompatibility = 1.8
+sourceCompatibility = 1.7
```
Comment (AndyMacDroo) [inline build.gradle]: Presumed we wanted to enforce compilation to JDK 1.7. Let me know if not and I'll swap back.

File: src/main/java/de/dmi3y/behaiv/tools/DataMappingUtils.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
+package de.dmi3y.behaiv.tools;
+
+import org.apache.commons.lang3.ArrayUtils;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public final class DataMappingUtils
```
Comment (AndyMacDroo) [inline src/main/java/de/dmi3y/behaiv/tools/DataMappingUtils.java]: New utility class here just for the sake of tidiness.

File: src/test/java/de/dmi3y/behaiv/session/CaptureSessionTest.java [hunk 3]
```diff
@@ -20,7 +20,7 @@
 
     @Before
     public void setUp() throws Exception {
-        session = new CaptureSession(Collections.singletonList(new TestProvider(new String[]{"name1", "name2"}, new Double[]{1.0})));
+        session = new CaptureSession(Collections.<Provider>singletonList(new TestProvider(new String[]{"name1", "name2"}, new Double[]{1.0})));
```
Comment (AndyMacDroo) [inline src/test/java/de/dmi3y/behaiv/session/CaptureSessionTest.java]: Yay for JDK 1.7 :joy:

**Comment by AndyMacDroo** [comment]
One thing that might be worth changing, probably not for now. There's a lot of use of the ArrayList concrete class and it might be better to switch method signatures around just to use the List interface - makes for easier / cleaner unit test creation.

**Comment by dmi3coder** [comment]
Thank you! Looks really great. Works on android below 21. I found some of my flaws while debugging :)
Yeah I was working fast to bring the prototype to live, now it's more of making the whole project stable enough for production 
Thank you again for bringing 0.4 version earlier by 5 days :)

**Comment by AndyMacDroo** [comment]
Thanks @dmi3coder - I'll keep an eye on the project and see what else I can help with in the future :)

**Pull request merged**